include/oneapi/dpl/pstl/hetero/numeric_impl_hetero.h

```diff
@@ -108,19 +121,210 @@ __pattern_transform_reduce(_ExecutionPolicy&& __exec, _ForwardIterator __first,
     return __res;
 }
 
+template< bool _Inclusive>
+struct __single_group_scan
+{
+    template<typename _Policy, typename _InRng, typename _OutRng, typename _InitType, typename _BinaryOperation, typename _UnaryOp>
+    static void __launch_dynamic_bounds_scan(_Policy const & __policy, _InRng __in, _OutRng __out, ::std::size_t __n, _InitType __init, const _BinaryOperation& __bin_op, const _UnaryOp& __unary_op, ::std::uint32_t __wg_size)
+    {
+        using _RangeValueType = decltype(*__in.begin());
+        using _ValueType = decltype(__bin_op(std::declval<_RangeValueType>(), std::declval<_RangeValueType>()));
+
+        using _CustomName = typename _Policy::kernel_name;
+        using _GroupScanKernel =
+                     oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_generator<__par_backend_hetero::__scan_single_wg_dynamic_kernel, _CustomName, _BinaryOperation, _InRng, _OutRng>;
+
+        ::uint32_t __elems_per_item = __par_backend_hetero::__ceiling_div(__n, __wg_size);
+        ::uint32_t __elems_per_wg = __elems_per_item * __wg_size;
+
+        auto __event = __policy.queue().submit([&](sycl::handler& __hdl) {
+            oneapi::dpl::__ranges::__require_access(__hdl, __in, __out);
+
+            auto __lacc =  __dpl_sycl::__local_accessor<_ValueType>(sycl::range<1>{__elems_per_wg}, __hdl);
+
+            __hdl.parallel_for<_GroupScanKernel>(sycl::nd_range<1>(__wg_size, __wg_size), [=](sycl::nd_item<1> __self_item) {
+                const auto& __group = __self_item.get_group();
+                const auto __item_id = __self_item.get_local_linear_id();
+
+                for (uint16_t __i = 0; __i < __elems_per_item; ++__i)
+                {
+                   auto __global_idx = __i * __wg_size + __item_id;
+                   __lacc[__global_idx] = __unary_op(__global_idx < __n ? __in[__global_idx] : unseq_backend::__known_identity<_BinaryOperation, _ValueType>);
```

Comment:
I have been keeping  in mind long time an idea about usage of a kind lazy range transformation (transfor_view or oneapi::dpl::__ranges::transform_view_simple) instead of explicit code for unary operation processed.
(https://github.com/oneapi-src/oneDPL/blob/main/include/oneapi/dpl/pstl/utils_ranges.h#L312)